kotlin-inject-compiler/test/src/test/kotlin/me/tatarka/inject/test/RoundsTest.kt

```diff
@@ -263,4 +265,45 @@ class RoundsTest {
             doesNotContain("Cannot provide", "as it is already provided")
         }
     }
-}
\ No newline at end of file
+
+  @ParameterizedTest
+  @EnumSource(Target::class)
+  fun scoped_provides_funs_that_have_the_same_return_type_with_different_type_args_does_not_conflict(target: Target) {
```

Comment:
Since this is a happy-path test, it can go it integration-tests instead.

Reply:
Updated